pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/RealtimeProvisioningHelperCommand.java

```diff
@@ -278,6 +280,7 @@ public boolean execute()
     displayResults(memoryEstimator.getConsumingMemoryPerHost(), numHosts, numHours);
     LOGGER.info("\nTotal number of segments queried per host (for all partitions)");
     displayResults(memoryEstimator.getNumSegmentsQueriedPerHost(), numHosts, numHours);
+    memoryEstimator.cleanup();
```

Comment:
Let's not make this an exposed API. The estimator should do the correct thing and everything that is necessary to ensure cleanup

Reply:
Good point. I agree.
Note that there are two ways segments are generated: 
1. Directly from SegmentGenerator.generate()
2. Implicitly in MemoryEstimator.estimateMemory()
For case 2, as you mentioned, we should clean it automatically and not delegate that to the caller of estimateMemory.
For case 1, it's up to the caller if they want to keep or delete the generated segment. So SegmentGenerator doesn't delete the segment automatically.